### PR TITLE
Add Knowledge Graph RAG Pipeline and QA System Examples

### PR DESCRIPTION
**Pull request by PulkitMishra**
This pull request adds two new examples to showcase Indexify's capabilities in building knowledge graph-based applications:

1. Knowledge Graph RAG (Retrieval-Augmented Generation) Pipeline:
   - Demonstrates entity extraction, relationship building, and knowledge graph construction
   - Integrates with Neo4j for graph storage and Sentence Transformers for text embedding
   - Provides a complete workflow from text input to graph storage and embedding generation

2. Knowledge Graph Question Answering (QA) System:
   - Showcases natural language question processing and conversion to Cypher queries
   - Utilizes Neo4j for graph querying and Google's Gemini AI for language tasks
   - Implements a fallback mechanism for robust query handling

Key changes:
- Added new directories under `examples/`: `knowledge_graph_generation/` and `knowledge_graph_qa_system/`
- Each new example includes a README.md with setup instructions and usage guidelines
- Updated the main README.md to include links to the new examples
- Implemented Indexify graphs for both knowledge graph workflows
- Added requirements.txt files for each new example

### CODE REVIEW

File: examples/knowledge_graph_qa_system/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+indexify
+neo4j
+google-generativeai
```
Comment (adithyaakrishna) [inline examples/knowledge_graph_qa_system/requirements.txt:3]:
```suggestion
google-generativeai

```

File: examples/knowledge_graph_qa_system/workflow.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        answer_result = client.graph_outputs(graph.name, invocation_id, "generate_answer")
+        if answer_result:
+            answer = answer_result[0]
+            logging.info(f"Generated Answer: {answer.text}")
+        else:
+            logging.warning("No answer generated")
+
+        logging.info("Workflow completed successfully!")
+    except Exception as e:
+        logging.error(f"An error occurred during execution: {str(e)}")
```
Comment (adithyaakrishna) [inline examples/knowledge_graph_qa_system/workflow.py:197]:
```suggestion
        logging.error(f"An error occurred during execution: {str(e)}")

```

File: examples/knowledge_graph_generation/workflow.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+    embeddings_result = client.graph_outputs(graph_name, invocation_id, "generate_embeddings")
+    if embeddings_result:
+        embedding = json.loads(embeddings_result[0].metadata['embedding'])
+        logging.info(f"Embeddings generated. First 5 values: {embedding[:5]}")
+    else:
+        logging.warning("No embeddings result")
+
+if __name__ == "__main__":
+    main()
```
Comment (adithyaakrishna) [inline examples/knowledge_graph_generation/workflow.py:232]:
```suggestion
    main()

```

**Review by diptanu — CHANGES_REQUESTED** [review]

**Comment by PulkitMishra** [comment]
hey @diptanu combined the two into one, as for avoiding loading twice i think instead of a hacky approach we should fix the sdk to do be able to handle it in a better way like we discussed . 

also @adithyaakrishna eol is there in readme and python file but not in requirements.txt just to be consistent with rest of code

(sorry for the force pushed commits, i committed and pushed from inside examples folders and realised later)

File: README.md [hunk 4]
```diff
@@ -11,6 +11,8 @@ Some of the use-cases that you can use Indexify for -
 * [Scraping and Summarizing websites](examples/website_audio_summary/)
 * [PDF Documents Extraction and Indexing](examples/pdf_document_extraction/)
 * [Transcribing audio and summarization](examples/website_audio_summary/)
+* [Knowledge Graph RAG Pipeline](examples/knowledge_graph_generation/)
```
[the remark this answers is not in the document]
Comment (PulkitMishra) [reply README.md:14]: done

**Comment by diptanu** [comment]
@PulkitMishra Lets add a docker compose here so people can spin it up easily and play with it!

**Comment by PulkitMishra** [comment]
@diptanu should i do it as part of #918 ?

**Pull request merged**